server/sonar-server/src/main/java/org/sonar/server/db/migration/DatabaseMigration.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.db.migration;
+
+import javax.annotation.Nullable;
+import java.util.Date;
+
+public interface DatabaseMigration {
+  enum Status {
+    NONE, RUNNING, FAILED, SUCCEEDED
+  }
+
+  /**
+   * Starts the migration status and returns immediately.
+   * <p>
+   * Migration can not be started twice but calling this method wont raise an error.
+   * On the other hand, calling this method when no migration is needed will start the process anyway.
+   * </p>
+   * <p>
+   * <strong>Do not rename this method to {@code start} otherwise it will be called by the pico container</strong>
+   * </p>
+   */
+  void startIt();
```

Comment:
I should have written in the javadoc "This method should be named {@code start} but it can not be because it will be called by the pico container and will cause unwanted behavior"

`startMigration` has the "migration" word duplicated, it terms of object design, that's not so good. Currently, calling this method can be written `databaseMigration.startIt()`. Since I can not use `start`, I really don't know which is worse of `startIt` and `startMigration` :(